src/librustdoc/core.rs

```diff
@@ -530,6 +531,16 @@ crate fn run_global_ctxt(
         module_trait_cache: RefCell::new(FxHashMap::default()),
         cache: Cache::default(),
     };
+
+    // Small hack to force the Sized trait to be present.
+    //
+    // Note that in case of `#![no_core]`, the trait is not available...
+    if let Some(sized_trait_did) = ctxt.tcx.lang_items().sized_trait() {
+        let mut sized_trait = build_external_trait(&ctxt, sized_trait_did);
+        sized_trait.is_auto = true;
+        ctxt.external_traits.borrow_mut().insert(sized_trait_did, sized_trait);
```

Comment:
You have unique access to `ctxt`, you don't need to use interior mutability.
```suggestion
        ctxt.external_traits.get_mut().insert(sized_trait_did, sized_trait);
```

Reply:
```
error[E0596]: cannot borrow data in an `Rc` as mutable
   --> src/librustdoc/core.rs:541:9
    |
541 |         ctxt.external_traits.get_mut().insert(sized_trait_did, sized_trait);
    |         ^^^^^^^^^^^^^^^^^^^^ cannot borrow as mutable
```

Reply:
Oh boo, ok. I think @camelid was going to make a PR getting rid of the Rc sometime soon.

Reply:
This one is hard to get rid of because it's shared between `DocContext` and `clean::Crate`. `clean::Crate` only seems to use it as a bridge between `DocContext` and `Cache`

https://github.com/rust-lang/rust/blob/9b471a3f5fe57e5c6e08acf665f2094422415a3d/src/librustdoc/clean/types.rs#L58-L60

`Cache` then `take`s the data

https://github.com/rust-lang/rust/blob/master/src/librustdoc/formats/cache.rs#L164

I think we'll need bigger architectural changes, like passing the field directly to `Cache`, in order to get rid of the intermediary. Doing that should make the code clearer and reduce the potential for sneaky bugs like giving the field to `Cache` via `Crate` and then trying to use the `DocContext` version later and finding nothing in it.